File: back/app/schemas/interview_booking.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field
from pydantic.alias_generators import to_camel


class CaseModel(BaseModel):
    model_config = ConfigDict(
        alias_generator=to_camel,
        populate_by_name=True,
        from_attributes=True,
    )


class InterviewBookingCreate(CaseModel):
    candidate_id: int = Field(..., gt=0)
    slot_id: int = Field(..., gt=0)


class InterviewBookingCancelRequest(CaseModel):
    candidate_id: int = Field(..., gt=0)


class AvailableInterviewSlotResponse(CaseModel):
    slot_id: int
    interview_round: str
    interview_starts_at: datetime
    interview_ends_at: datetime
    interview_location: str | None
    remaining_capacity: int
    interviewer_names: list[str] = Field(default_factory=list)


class InterviewBookingResponse(CaseModel):
    booking_id: int
    candidate_id: int
    slot_id: int
    interview_starts_at: datetime
    interview_ends_at: datetime
    interview_location: str | None
    created_at: datetime


class ActiveBookingSummaryResponse(CaseModel):
    """직무별 활성 예약 batch 조회 응답 (지원자 → 어떤 슬롯에 배정되어 있는지)"""

    booking_id: int
    candidate_id: int
    slot_id: int
    position_id: int | None
    position_name: str | None
    interview_round: str
    interview_starts_at: datetime
    interview_ends_at: datetime
    interview_location: str | None
    booked_at: datetime

File: front/types/interviewBooking.ts
export interface AvailableInterviewSlot {
  slotId: number;
  interviewRound: string;
  interviewStartsAt: string;
  interviewEndsAt: string;
  interviewLocation: string | null;
  remainingCapacity: number;
  interviewerNames: string[];
}

export interface InterviewBookingPayload {
  candidateId: number;
  slotId: number;
}

export interface InterviewBookingCancelPayload {
  candidateId: number;
}

export interface InterviewBookingResponse {
  bookingId: number;
  candidateId: number;
  slotId: number;
  interviewStartsAt: string;
  interviewEndsAt: string;
  interviewLocation: string | null;
  createdAt: string;
}

export interface InterviewBookingMutationResponse {
  message: string;
}

/**
 * 직무에 걸린 활성 booking 일괄 조회 응답.
 * 같은 직무 지원자 배정 UI에서 "다른 슬롯에 배정됨" 라벨을 그리고
 * 해당 슬롯 일자로 점프시키는 데 사용됩니다.
 */
export interface ActiveBookingSummary {
  bookingId: number;
  candidateId: number;
  slotId: number;
  positionId: number | null;
  positionName: string | null;
  interviewRound: string;
  interviewStartsAt: string;
  interviewEndsAt: string;
  interviewLocation: string | null;
  bookedAt: string;
}
